src/VisualStudio/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.VisualStudio.Test.Utilities.Common;
+
+namespace Roslyn.VisualStudio.Test.Utilities.InProcess
+{
+    public class Editor_InProc : InProcComponent
+    {
+        private static readonly Guid IWpfTextViewId = new Guid("8C40265E-9FDB-4F54-A0FD-EBB72B7D0476");
+
+        private Editor_InProc() { }
+
+        public static Editor_InProc Create()
+        {
+            return new Editor_InProc();
+        }
+
+        private static ITextView GetActiveTextView()
+        {
+            return GetActiveTextViewHost().TextView;
+        }
+
+        private static IVsTextView GetActiveVsTextView()
+        {
+            IVsTextView vsTextView = null;
+
+            var vsTextManager = GetGlobalService<SVsTextManager, IVsTextManager>();
+
+            var hresult = vsTextManager.GetActiveView(fMustHaveFocus: 1, pBuffer: null, ppView: out vsTextView);
+            Marshal.ThrowExceptionForHR(hresult);
+
+            return vsTextView;
+        }
+
+        private static IWpfTextViewHost GetActiveTextViewHost()
+        {
+            // The active text view might not have finished composing yet, waiting for the application to 'idle'
+            // means that it is done pumping messages (including WM_PAINT) and the window should return the correct text view
+            WaitForApplicationIdle();
+
+            var activeVsTextView = (IVsUserData)GetActiveVsTextView();
+
+            object wpfTextViewHost;
+            var hresult = activeVsTextView.GetData(IWpfTextViewId, out wpfTextViewHost);
+            Marshal.ThrowExceptionForHR(hresult);
+
+            return (IWpfTextViewHost)wpfTextViewHost;
+        }
+
+        private static void ExecuteOnActiveView(Action<ITextView> action)
+        {
+            InvokeOnUIThread(() =>
+            {
+                var view = GetActiveTextView();
+                action(view);
+            });
+        }
+
+        private static T ExecuteOnActiveView<T>(Func<ITextView, T> action)
+        {
+            return InvokeOnUIThread(() =>
+            {
+                var view = GetActiveTextView();
+                return action(view);
+            });
+        }
+
+        public void Activate()
+        {
+            GetDTE().ActiveDocument.Activate();
+        }
+
+        public string GetText()
```

Comment:
Was the plan for Get* functions to be named like this because they're doing marshalling under the covers?